refactor(cli): replace custom confirmation prompts with prompts library

ArnavK-09 opened the pull request:
- Removed the custom askConfirmation function and integrated the prompts library for user confirmations.
- Updated the initialization process to use prompts for confirming project setup in the current directory.
- Enhanced the update check process to utilize prompts for user decisions on updating the CLI version.

ArnavK-09 committed:
refactor(cli): replace custom confirmation prompts with prompts library

- Removed the custom askConfirmation function and integrated the prompts library for user confirmations.
- Updated the initialization process to use prompts for confirming project setup in the current directory.
- Enhanced the update check process to utilize prompts for user decisions on updating the CLI version.

seveibar commented on cli/init/register.ts:
```diff
@@ -6,10 +6,8 @@ import { generateTsConfig } from "lib/shared/generate-ts-config"
-  checkForTsciUpdates,
-} from "lib/shared/check-for-cli-update"
+import { checkForTsciUpdates } from "lib/shared/check-for-cli-update"
+import prompt from "prompts"
```
```suggestion
import prompt from "lib/prompts"
```

seveibar commented on lib/shared/check-for-cli-update.ts:
```diff
@@ -7,6 +7,7 @@ import { program } from "cli/main"
 import semver from "semver"
 import { version as pkgVersion } from "../../package.json"
 import kleur from "kleur"
+import prompt from "prompts"
```
```suggestion
import prompt from "lib/prompts"
```

seveibar reviewed (approved):
import from `lib/prompts` to enable some CI automation

ArnavK-09 committed:
refactor(cli): standardize prompt usage by importing from prompts library

- Replaced direct imports of the prompts library in the CLI initialization and update check processes.
- Ensured consistent usage of the prompts library for user confirmations across the codebase.
